tango_ros_common/tango_ros_native/src/tango_ros_node.cpp

```diff
@@ -404,7 +542,33 @@ void TangoRosNode::OnPoseAvailable(const TangoPoseData* pose) {
 }
 
 void TangoRosNode::OnPointCloudAvailable(const TangoPointCloud* point_cloud) {
-  if (publisher_config_.publish_point_cloud && point_cloud->num_points > 0
+  if (publisher_config_.publish_laser_scan && point_cloud->num_points > 0
+      && laser_scan_available_mutex_.try_lock()) {
+    laser_scan_.angle_min = LASER_SCAN_ANGLE_MIN;
+    laser_scan_.angle_max = LASER_SCAN_ANGLE_MAX;
+    laser_scan_.angle_increment = LASER_SCAN_ANGLE_INCREMENT;
+    laser_scan_.time_increment = LASER_SCAN_TIME_INCREMENT;
+    laser_scan_.scan_time = LASER_SCAN_SCAN_TIME;
+    laser_scan_.range_min = LASER_SCAN_RANGE_MIN;
+    laser_scan_.range_max = LASER_SCAN_RANGE_MAX;
+    // Determine amount of rays to create.
+    uint32_t ranges_size = std::ceil((laser_scan_.angle_max - laser_scan_.angle_min)
+                                     / laser_scan_.angle_increment);
+    // Laser scan rays with no obstacle data will evaluate to infinity.
+    laser_scan_.ranges.assign(ranges_size, std::numeric_limits<double>::infinity());
+    if (publisher_config_.publish_point_cloud && point_cloud_available_mutex_.try_lock()) {
+      toPointCloud2AndLaserScan(*point_cloud, time_offset_, laser_scan_min_height_,
+                               laser_scan_max_height_, &point_cloud_, &laser_scan_);
+      point_cloud_.header.frame_id = toFrameId(TANGO_COORDINATE_FRAME_CAMERA_DEPTH);
+      point_cloud_available_.notify_all();
+      point_cloud_available_mutex_.unlock();
+    } else {
```

Comment:
I believe this is related to having a function that publishes both at the same time. 
The else clause below publishes the laser scan alone. This part will be resolved once a decision is taken about the function `toPointCloud2AndLaserScan`